src/Modal.js

```diff
@@ -166,59 +198,70 @@ class Modal extends React.Component {
 
   renderChildren() {
     const {
-      className,
       wrapClassName,
       modalClassName,
       backdropClassName,
-      contentClassName,
       cssModule,
       isOpen,
-      size,
-      backdrop,
-      children,
-      ...attributes
-    } = omit(this.props, ['toggle', 'keyboard', 'onEnter', 'onExit', 'zIndex']);
+      backdrop
+    } = this.props;
+
+    const modalAttributes = {
+      onClickCapture: this.handleBackdropClick,
+      onKeyUp: this.handleEscape,
+      style: { display: 'block' },
+      tabIndex: '-1'
+    };
+
+    if (this.hasTransition()) {
+      const modalTransitionTimeout = 300;
+      const backdropTransitionTimeout = 150;
```

Comment:
I've pushed another commit after seeing issue https://github.com/reactstrap/reactstrap/issues/352. That person wants the ability to separately configure appear vs. enter vs. leave timeouts.

It's a lot of new props, but all optional ones. What do you think?
https://github.com/reactstrap/reactstrap/pull/339/commits/ab93b7dd1ec26c0ab29fc6b6a0d03dbc6ca7e01c